x/incentive/keeper/btc_staking_gauge.go

```diff
@@ -61,28 +68,47 @@ func (k Keeper) RewardBTCStaking(ctx context.Context, height uint64, dc *ftypes.
 
 		// reward the finality provider with commission
 		coinsForCommission := types.GetCoinsPortion(coinsForFpsAndDels, *fp.Commission)
-		k.accumulateRewardGauge(ctx, types.FINALITY_PROVIDER, fp.GetAddress(), coinsForCommission)
+
+		// add FP direct rewards from fee collector (goes entirely to FP)
+		coinsForFpDirect := fpDirectGauge.GetCoinsPortion(fpPortion)
+		totalCoinsForFp := coinsForCommission.Add(coinsForFpDirect...)
+
+		k.accumulateRewardGauge(ctx, types.FINALITY_PROVIDER, fp.GetAddress(), totalCoinsForFp)
 
 		// reward the rest of coins to each BTC delegation proportional to its voting power portion
 		coinsForBTCDels := coinsForFpsAndDels.Sub(coinsForCommission...)
 		if err := k.AddFinalityProviderRewardsForBtcDelegations(ctx, fp.GetAddress(), coinsForBTCDels); err != nil {
 			panic(fmt.Errorf("failed to add fp rewards for btc delegation %s at height %d: %w", fp.GetAddress().String(), height, err))
 		}
 	}
-	// TODO: prune unnecessary state (delete BTCStakingGauge after the amount is used)
+
+	// Prune gauges after rewards are distributed to free up storage
+	// k.pruneBTCStakingGauge(ctx, height) // this is used in e2e tests and queries, need to decide what to do
```

Comment:
In the software upgrade test, we're querying the BTCStakingGauges to calculate the expected rewards.
Also, there's the module query `BTCStakingGauge`. A few questions here:
- do we want to still prune these and update the e2e test?
- should we add also a query for the  `FPDirectGauge`?
If we're pruning, the queries will work only for those heights where there haven't been any rewards distribution.
Is there a specific reason why we may want to keep the module query?

Reply:
Other approach, if we want to have these numbers, is emitting an event. So the data will be available in nodes that did not prune the block

Reply:
sg, will add the events, so we can have these values for all blocks (the ones that pruned the gauge and the ones that don't)